otlp/translator.go

```diff
@@ -386,16 +389,31 @@ func getTagFromSpanKind(spanKind otlptrace.Span_SpanKind) (model.KeyValue, bool)
 	}, true
 }
 
+func handleDeprecatedStatusCode(status *otlptrace.Status) {
+	// Apply rules defined by the deprecation transition
+	// https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/trace/v1/trace.proto#L254
+	if status.Code == otlptrace.Status_STATUS_CODE_UNSET && status.DeprecatedCode != otlptrace.Status_DEPRECATED_STATUS_CODE_OK {
+		status.Code = otlptrace.Status_STATUS_CODE_ERROR
```

Comment:
Are we allowed to modify? This is used in SAPM exporter I think, in which case it is not allowed.

Reply:
This is not used by SAPM exporter :)

Reply:
The public API is "body bytes" to "SAPM", so OTLP is an intermediate format not exposed to the users, we can change it.